types/k6/test/timers.ts

```diff
@@ -33,7 +33,16 @@ setInterval('functionRef', 5);
 // @ts-expect-error
 setInterval('functionRef', 5, 'arg');
 
+// @ts-expect-no-error
+const intervalId = setInterval(function(){}, 5);
+
 // clearInterval
 
+// @ts-expect-no-error
+clearInterval(intervalId)
+
+// @ts-expect-no-error
```

Comment:
```suggestion
// @ts-expect-error
```
Here we do not use the result of a setInterval so we *do* expect an error

Reply:
It will be nicer if it does cause an error as you shouldn't really ever use "just" a number you should always use a value you got from `setInterval` but it is okay - we can fix this when somebody who knows how to make that happen finds this ;)

Reply:
> It will be nicer if it does cause an error as you shouldn't really ever use "just" a number you should always use a value you got from `setInterval` but it is okay - we can fix this when somebody who knows how to make that happen finds this ;)

What you want to achieve is called "Nominal Type", however typescript doesn't support it. There is a workaround as described here: https://dnlytras.com/blog/nominal-types . But I've not seen a project that makes use of it. This is the way things go with typescript 😕